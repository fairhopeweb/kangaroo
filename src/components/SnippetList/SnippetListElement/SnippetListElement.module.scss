@import "@blueprintjs/core/lib/scss/variables";

.root {
  height: var(--snippet-list-element-height);

  &.active {
    background-image: var(--active-snippet);

    .tag {
      box-shadow: 0 -0.5px 0px 1px var(--active-snippet-tag-border);
    }
  }
}

.content {
  position: relative;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  height: 100%;
  padding: 5px 15px 5px 15px;
  border-image: var(--divider) 1;
  border-bottom-style: solid;
  border-bottom-width: 1px;
}

.tag {
  position: absolute;
  width: 10px;
  height: 10px;
  right: 5px;
  border-radius: 50%;
  box-shadow: 0 -0.5px 0px 1px var(--background-color-secondary);
}

.sourceIcon {
  padding: 5px 10px;
}
