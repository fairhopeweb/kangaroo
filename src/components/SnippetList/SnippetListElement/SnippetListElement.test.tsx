import { configure, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import renderer from 'react-test-renderer';

import SnippetListElement from './SnippetListElement';

configure({ adapter: new Adapter() });

describe('<SnippetListElement />', () => {
  const mockProps = {
    snippetId: 1,
    snippetTags: '1,2,3',
    snippetTitle: 'Test',
    currentlySelectedId: 1,
    onChangeSnippet: jest.fn(),
    onContextMenu: jest.fn(),
  };

  it('renders without crashing', () => {
    shallow(<SnippetListElement {...mockProps} />);
  });

  it('matches snapshot', () => {
    const wrapper = renderer.create(<SnippetListElement {...mockProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});
