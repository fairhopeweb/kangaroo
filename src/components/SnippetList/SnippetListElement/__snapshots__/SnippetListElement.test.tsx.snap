// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnippetListElement /> matches snapshot 1`] = `
<div
  className="root active"
  onClick={[Function]}
  onContextMenu={[Function]}
>
  <div
    className="content"
  >
    <span
      className="bp3-text-overflow-ellipsis"
    >
      Test
    </span>
    <div
      className="tag"
      style={
        Object {
          "background": "#F55656",
          "top": "6px",
        }
      }
    />
    <div
      className="tag"
      style={
        Object {
          "background": "#F29D49",
          "top": "14px",
        }
      }
    />
    <div
      className="tag"
      style={
        Object {
          "background": "#F2B824",
          "top": "22px",
        }
      }
    />
  </div>
</div>
`;
